### ISSUE TITLE
Pagination doesnt work on a collection page

### ISSUE THREAD

**Issue by hosnas** [sverrirs/jekyll-paginate-v2#84]
I want to use some of my collection pages to show pagination of other collections. but it seems this plugin doesnt support this functionality. Any reason why? and any chance it could be considered as a feature request?

**Comment by woutrbe** [comment]
I'm currently facing the same issue, even though I've added
`pagination:
   enabled: true`
to the front matter of all my collection pages, the paginator won't pick it up.

Just to clarify, my site structure is as follows:
`- _products
  - product1.md
  - product2.md
- _types
  - electronis.md
  - outdoor.md`

`_products` have the following front matter
`
types:
  - electronics
  - outdoor
`

In my `_types` template I can easily get all the products
`{% assign products = site.products | where_exp: "product", "product.types contains page.slug" %}`

But how do I get pagination to work? None of the solutions or examples seem to work. I'm getting slightly confused as this doesn't seem like a unusual use-case.

**Comment by sverrirs** [comment]
Hi, could you please provide me with a repro of this issue in a git repo somewhere or a zip file that I can test out. This may be a configuration that I hadn't anticipated and it is impossible to figure out why this is going wrong without being able to debug the exact issue :)

**Comment by woutrbe** [comment]
I just created a new project to show my configuration: https://github.com/woutrbe/jekyll-pagination
Basically I want to paginate all category pages under http://127.0.0.1:4000/categories

But a few questions:
1. How do I do this for categories where I have a custom post list
2. Even with pagination enabled in the layout, jekyll complains about `Is enabled, but I couldn't find any pagination page.`

Let me know if the example is clear enough.

**Comment by hosnas** [comment]
@sverrirs  as per [here](https://github.com/sverrirs/jekyll-paginate-v2/blob/7a15647777b12795515ffacbf3b9c09bc9b7f452/lib/jekyll-paginate-v2/generator/paginationGenerator.rb#L65)  collection docs aren't considered as a candidate for pagination. it means pagination can be only enabled in pages. I hope this would be enhanced to include collection docs too.

**Comment by renshuki** [comment]
@hosnas Isn't this already supported by [AutoPages](https://github.com/sverrirs/jekyll-paginate-v2/blob/master/README-AUTOPAGES.md) addon?

**Comment by hosnas** [comment]
@renshuki I don't think so. How?

**Comment by joshdick** [comment]
I'm having the same issue.

The collection I'm trying to paginate is called `microblog`. In the example below, `/_microblog/index.html` is not seen by `jekyll-paginate-v2` even though it has `pagination` configuration in its front matter.

I don't see the `Is enabled, but I couldn't find any pagination page.` warning unless I explicitly enable pagination in the root `_config.yaml`, but regardless of what `_config.yaml` says, the pagination configuration in the front matter in `/_microblog/index.html` appears to be completely ignored.

Here's my setup, all relative to a top-level directory:

`/Gemfile`:

```gemfile
source 'https://rubygems.org'
gem 'jekyll'
gem 'jekyll-paginate-v2'
```

`/_config.yml` (note that there's no `pagination`-related configuration here):

```yaml
collections:
  microblog:
    output: true
exclude: ['Gemfile', 'Gemfile.lock']
kramdown:
  entity_output: numeric
sass:
  style: compressed
  sass_dir: _sass
autopages:
  enabled: true
  collections:
    layouts:
      - microblog/index.html
plugins:
- jekyll-paginate-v2
```

`/_microblog/posts`: (Contains a bunch of Markdown files)

`/_microblog/index.html`:

```html
---
layout: default
title: Microblog
pagination:
  enabled: true
  collection: microblog
---
<br />

<h1>Microblog</h1>

{% for post in paginator.posts %}
  <h1>{{ post.content }}</h1>
{% endfor %}

{% if paginator.total_pages > 1 %}
<div class="pagination">
  {% if paginator.previous_page %}
    <a href="{{ paginator.previous_page_path | prepend: site.baseurl | replace: '//', '/' }}">&laquo; Prev</a>
  {% else %}
    <span>&laquo; Prev</span>
  {% endif %}

  {% for page in (1..paginator.total_pages) %}
    {% if page == paginator.page %}
      <em>{{ page }}</em>
    {% elsif page == 1 %}
      <a href="{{ paginator.previous_page_path | prepend: site.baseurl | replace: '//', '/' }}">{{ page }}</a>
    {% else %}
      <a href="{{ site.paginate_path | prepend: site.baseurl | replace: '//', '/' | replace: ':num', page }}">{{ page }}</a>
    {% endif %}
  {% endfor %}

  {% if paginator.next_page %}
    <a href="{{ paginator.next_page_path | prepend: site.baseurl | replace: '//', '/' }}">Next &raquo;</a>
  {% else %}
    <span>Next &raquo;</span>
  {% endif %}
</div>
{% endif %}

<!-- Unrelated content follows -->
```

I hope that's a clear example that demonstrates the issue, but I'm happy to clarify anything if needed.

@sverrirs

**Comment by spero-cnet** [comment]
@joshdick you might also need to add a string filter to remove /index.html if your _config.yml is set to expressionless urls (pretty)- which paginate v2 does not seem to honour.

`
{% if paginator.total_pages > 1 %}
<div class="row">
	<div class="col-sm-12 text-center m-t-40 m-b-40">
		<ul class="pagination font-alt">
			{% if paginator.previous_page %}
			<li><a href="{{ paginator.previous_page_path | prepend: site.baseurl }}"><i class="fa fa-angle-left"></i></a></li>
			{% endif %}
			{% if paginator.page_trail %}
				{% for trail in paginator.page_trail %}
					<li {% if page.url == trail.path %}class="active"{% endif %}>
						<a href="{{ trail.path | prepend: site.baseurl | remove: "index.html" }}" title="{{trail.title}}">{{ trail.num }}</a>
					</li>
				{% endfor %}
			{% endif %}
			{% if paginator.next_page %}
				<li><a href="{{ paginator.next_page_path | prepend: site.baseurl }}"><i class="fa fa-angle-right"></i></a></li>
			{% endif %}
		</ul>
	</div>
</div>
{% endif %}
`

**Comment by joshdick** [comment]
@spero-cnet That indeed happened to me and your code did the trick, thanks for the tip.

**Comment by vcavallo** [comment]
> Figured it out.
> 
> Based on my example above, my existing `/_microblog/posts` folder needed to only contain the Markdown files, but I needed to move `index.html` to `/microblog/index.html` to get pagination to work (as opposed to `/_microblog/index.html` where it was before -- no underscore.)
> 
> Other resources not mentioned in my example above that used to be in `/_microblog` (RSS feed, etc.) also needed to be moved to `/microblog` so that `jekyll-paginate-v2` wouldn't try to also paginate them.
> 
> tl;dr posts need to be inside `/_collectionName/posts`, but the page to be paginated needs to live in `/collectionName`.

Thank you thank you. I was having the same issue and searching all over this repo was not helping. Your comment was the only thing that cleared it up (well, your comment _and_ taking a lunch break before deep frustration set in :) )

**Comment by joshdick** [comment]
@vcavallo You're welcome!

**Comment by pixelicous** [comment]
> Figured it out.
> 
> Based on my example above, my existing `/_microblog/posts` folder needed to only contain the Markdown files, but I needed to move `index.html` to `/microblog/index.html` to get pagination to work (as opposed to `/_microblog/index.html` where it was before -- no underscore.)
> 
> Other resources not mentioned in my example above that used to be in `/_microblog` (RSS feed, etc.) also needed to be moved to `/microblog` so that `jekyll-paginate-v2` wouldn't try to also paginate them.
> 
> tl;dr posts need to be inside `/_collectionName/posts`, but the page to be paginated needs to live in `/collectionName`.

Thanks.. was stumbling upon this as well.. it did seem from #104 that it could be fixed in other ways, but couldnt get that to work

**Comment by bedlamzd** [comment]
Faced same issue and found this thread.
It's a solution, yes, but isn't it kind of redundant to make additional directory only for pagination page? Also it keeps related files separate which I would consider unwanted.

I'm trying to make site for my lab and I'm new to Jekyll (but tried Pelican before). ([repo](https://github.com/mt-lab/MT.Site))

Idea is that we have some long-term projects (this projects I consider as collection) and each project should have it's own page with some general description and current state. Also each project can have it's own posts/news. So I want to list them on project page as pagination. So I would have on each project page kind of a blogroll with updates and posts related to this project.

For it not to make any mess and be organized I would like to have all project related files in dedicated directory.

I thought about making each project a category but with this setup I couldn't figure out how to make general page for project. As I understand category can't have index.html page with description and content.
